tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -0,0 +1,122 @@
+func createHostFile(t *testing.T, ip, password string) string {
+	hostFile, err := ioutil.TempFile("", "testWSU")
+	defer hostFile.Close()
+	require.NoErrorf(t, err, "Could not make temporary file: %s", err)
+
+	_, err = hostFile.WriteString(fmt.Sprintf(`[win]
+%s ansible_password='%s'
+
+[win:vars]
+ansible_user=Administrator
+cluster_address=%s
+ansible_port=5986
+ansible_connection=winrm
+ansible_winrm_server_cert_validation=ignore`, ip, password, clusterAddress))
+	require.NoError(t, err, "Could not write to host file: %s", err)
+	return hostFile.Name()
+}
+
+// TestWSU creates a Windows instance, runs the WSU, and then runs a series of tests to ensure all expected
+// behavior was achieved
+func TestWSU(t *testing.T) {
+	// TODO: Check if other cloud provider credentials are available
+	if awscredentials != "" {
+		createAWSWindowsInstance(t)
+	} else {
+		t.Fatal("No cloud provider credentials available")
+	}
+	defer cloud.DestroyWindowsVMs()
+	// In order to run the ansible playbook we create an inventory file:
+	// https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+	hostFilePath := createHostFile(t, createdInstanceCreds.GetIPAddress(), createdInstanceCreds.GetPassword())
+	cmd := exec.Command("ansible-playbook", "-vvv", "-i", hostFilePath, playbookPath)
+	out, err := cmd.CombinedOutput()
+	assert.NoError(t, err, "WSU playbook returned error: %s, with output: ", err, string(out))
+
+	// Ansible will copy files to a temporary directory with a path such as:
+	// C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible.z5wa1pc5.vhn\\
+	split := strings.Split(string(out), "C:\\\\Users\\\\Administrator\\\\AppData\\\\Local\\\\Temp\\\\ansible.")
+	require.True(t, len(split) > 1, "Could not find Windows temp dir: %s", out)
+	ansibleTmpDir = "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible." + strings.Split(split[1], "\"")[0]
+
+	t.Run("Files copied to Windows node", testFilesCopied)
+	// TODO: Once the WSU starts the WMCB and adds the node to the cluster, add check to see if the node is "Ready"
+}
+
+// testFilesCopied tests that the files we attempted to copy to the Windows host, exist on the Windows host
+func testFilesCopied(t *testing.T) {
+	expectedFileList := []string{"kubelet.exe", "worker.ign", "wmcb.exe"}
+	endpoint := winrm.NewEndpoint(createdInstanceCreds.GetIPAddress(), 5986, true, true,
+		nil, nil, nil, 0)
+	client, err := winrm.NewClient(endpoint, "Administrator", createdInstanceCreds.GetPassword())
+	require.NoErrorf(t, err, "Could not create winrm client: %s", err)
+
+	// Check if each of the files we expect on the Windows host are there
+	for _, filename := range expectedFileList {
+		fullPath := ansibleTmpDir + "\\" + filename
+		command := fmt.Sprintf("if not exist %s echo fail", fullPath)
+		buf := new(bytes.Buffer)
+		_, err := client.Run(command, buf, os.Stderr)
+		assert.NoError(t, err, "Error looking for %s: %s", fullPath, err)
+		assert.Emptyf(t, buf.String(), "Missing file: %s", fullPath)
```

Comment:
So you expect `buf` to be empty? Wouldn't it be the other way around if the file exists?

Reply:
We run the command `if not exist $FILENAME echo fail`, which will only write to stdout if the file doesn't exist.

Reply:
Please add comment stating that as it is not obvious.